lib/rules/no-tabs.js

```diff
@@ -21,25 +21,45 @@ module.exports = {
             category: "Stylistic Issues",
             recommended: false
         },
-        schema: []
+        schema: [],
+        fixable: "whitespace"
     },
 
     create(context) {
         return {
             Program(node) {
+                let totalLength = 0;
+
                 context.getSourceLines().forEach((line, index) => {
-                    const match = regex.exec(line);
+                    let match;
+
+                    // Intentional use of conditional assignment - .exec() returns null when it
+                    // cannot find more matches, breaking the loop
+                    // eslint-disable-next-line no-cond-assign
+                    while (match = regex.exec(line)) {
+                        const loc = {
+                            line: index + 1,
+                            column: match.index + 1
+                        };
 
-                    if (match) {
-                        context.report(
+                        context.report({
                             node,
-                            {
-                                line: index + 1,
-                                column: match.index + 1
-                            },
-                            "Unexpected tab character."
-                        );
+                            loc,
+                            message: "Unexpected tab character.",
+
+                            // This function needs to capture the current `match`
+                            // eslint-disable-next-line no-loop-func
```

Comment:
I believe this was a problem with `var` due to them being scoped to the function, not to current block. However, this rule uses `let` and `const` which are block-scoped and thus the `fix` function will see the intended value when called. Nevertheless, I have added tests for this.